ocaml/typing/printtyp.ml

```diff
@@ -1281,7 +1257,20 @@ let outcome_label : Types.arg_label -> Outcometree.arg_label = function
   | Optional l -> Optional l
   | Position l -> Position l
 
-let rec tree_of_typexp mode ty =
+let tree_of_mode mode l =
```

Comment:
```suggestion
(* If the [mode] is [Some m], returns a list containing the corresponding [out_mode] from
   the list [l].
   Precondition: If [mode] is [Some m], [m] is included as a first element in [l]. *)
let tree_of_mode (mode : 'mode option) (l : ('mode * out_mode) list) : out_mode list =
```

Reply:
After the type annotation, I believe the function is clear to read. The function is only 3 lines, after all.